content/docs/user-guide/merge-conflicts.md

```diff
@@ -0,0 +1,198 @@
+  - path: data/small.xml
+>>>>>>> branch
+    md5: 285af85d794bb57e5d09ace7209f3519
+  - path: src/prepare.py
+<<<<<<< HEAD
+    md5: a304afb96060aad90176268345e10355
+=======
+    md5: 0d2602f440d13372435d91d3bfa17652
+>>>>>>> branch
+  params:
+    params.yaml:
+      prepare.seed: 20170428
+      prepare.split: 0.2
+  outs:
+  - path: data/prepared
+    md5: 20b786b6e6f80e2b3fcf17827ad18597.dir
+```
+
+## \*.dvc
+
+There are three three types of such files that differ by the command that has
+created them.
+
+### dvc import
+
+Simply remove the hashes
+
+```
+<<<<<<< HEAD
+md5: 263395583f35403c8e0b1b94b30bea32
+=======
+md5: 520d2602f440d13372435d91d3bfa176
+>>>>>>> branch
+frozen: true
+deps:
+- path: get-started/data.xml
+  repo:
+    url: https://github.com/iterative/dataset-registry
+<<<<<<< HEAD
+    rev_lock: f31f5c4cdae787b4bdeb97a717687d44667d9e62
+=======
+    rev_lock: 06be1104741f8a7c65449322a1fcc8c5f1070a1e
+>>>>>>> branch
+outs:
+<<<<<<< HEAD
+- md5: a304afb96060aad90176268345e10355
+=======
+- md5: 35dd1fda9cfb4b645ae431f4621fa324
+>>>>>>>
+  path: data.xml
+```
+
+so you get
+
+```
+frozen: true
+deps:
+- path: get-started/data.xml
+  repo:
+    url: https://github.com/iterative/dataset-registry
+outs:
+- path: data.xml
+```
+
+and then run `dvc repro`.
+
+### dvc import-url
+
+Simply remove the hashes
+
+```
+<<<<<<< HEAD
+md5: 263395583f35403c8e0b1b94b30bea32
+=======
+md5: 520d2602f440d13372435d91d3bfa176
+>>>>>>> branch
+deps:
+<<<<<<< HEAD
+- etag: 696df35ad1161afbeb6ea667e5dd5dab-2861
+=======
+- etag: 266336746d8a06be1124153ff2156236-132
+>>>>>>> branch
+  path: s3://bucket/data.xml
+outs:
+<<<<<<< HEAD
+- md5: a304afb96060aad90176268345e10355
+=======
+- md5: 35dd1fda9cfb4b645ae431f4621fa324
+>>>>>>>
+  path: data.xml
+```
+
+so you get
+
+```
+deps:
+- path: s3://bucket/data.xml
+outs:
+- path: data.xml
+```
+
+and then run `dvc repro`.
+
+### dvc add
+
+In this case you'll get something that looks like
+
+```
+outs:
+<<<<<<< HEAD
+- md5: a304afb96060aad90176268345e10355
+=======
+- md5: 35dd1fda9cfb4b645ae431f4621fa324
+>>>>>>> branch
+  path: data.xml
+```
+
+If you decide to just pick one of the versions, you could just leave the md5 for
+the desired version and delete the second one:
+
+```
+outs:
+- md5: 35dd1fda9cfb4b645ae431f4621fa324
+  path: data.xml
+```
+
+or you could use `-x theirs` or `-x ours` option for `git merge`.
+
+But if you want to something more complex like merging the two files/directories
+together, then you'll need to run `dvc checkout data.xml` on both `HEAD` and
+`branch`, copy the data into separate locations (e.g. `data.xml.head` and
+`data.xml.branch`), then merge them by-hand and run `dvc add data.xml` for the
+resulting merged file/directory.
+
+#### Append-only directories
+
+If you have an "append-only" dataset, where people only add new
+files/directories, DVC provides a so-called
+[merge-driver](https://git-scm.com/docs/git-merge#Documentation/git-merge.txt-mergeltdrivergtname)
```

Comment:
Interesting, didn't know about those!

Maybe link to https://git-scm.com/docs/gitattributes#_defining_a_custom_merge_driver though?

Reply:
@jorgeorpinel That link is mentioned in the first link, so probably doesn't matter which one we keep.

Reply:
Well... It's not the exact same link. But I guess neither one defines "merge driver" directly so yeah, it's fine!